api/failure_detector.cc

```diff
@@ -8,9 +8,94 @@
 
 #include "failure_detector.hh"
 #include "api/api-doc/failure_detector.json.hh"
-#include "gms/failure_detector.hh"
 #include "gms/application_state.hh"
 #include "gms/gossiper.hh"
+#include "utils/bounded_stats_deque.hh"
+#include "log.hh"
+
+extern logging::logger apilog;
+
+namespace gms {
+
+class arrival_window {
```

Comment:
The only use is here:

```

    fd::get_endpoint_phi_values.set(r, [](std::unique_ptr<request> req) {
        std::map<gms::inet_address, gms::arrival_window> map;
        std::vector<fd::endpoint_phi_value> res;
        auto now = gms::arrival_window::clk::now();
        for (auto& p : map) {
            fd::endpoint_phi_value val;
            val.endpoint = p.first.to_sstring();
            val.phi = p.second.phi(now);
            res.emplace_back(std::move(val));
        }
        return make_ready_future<json::json_return_type>(res);
    });
```

Notice that `map` is not initialized, so it's empty. Before 0ef33b71bab2b15f8f42acea44fd4049fafe46e2, it was obtained from the gossiper gms::get_arrival_samples().

So you can just remove arrival_window altogether.

I don't think the logic of arrival_window belongs to api, it belongs to the failure detector. If there is no longer a failure detector, there should be no arrival_window.

Reply:
fixed in v2